services/worker/tests/job_runners/split/test_descriptive_stats.py

```diff
@@ -0,0 +1,263 @@
+            hf_datasets_cache=libraries_resource.hf_datasets_cache,
+        )
+
+    return _get_job_runner
+
+
+EXPECTED_STATS_CONTENT = {
+    "num_examples": 20,
+    "stats": [
+        {
+            "column_name": "class_label_column",
+            "column_type": "class_label",
+            "column_dtype": None,
+            "column_stats": {"nan_count": 0, "nan_prop": 0.0, "n_unique": 2, "frequencies": {"cat": 17, "dog": 3}},
+        },
+        {
+            "column_name": "class_label_nan_column",
+            "column_type": "class_label",
+            "column_dtype": None,
+            "column_stats": {"nan_count": 4, "nan_prop": 0.2, "n_unique": 3, "frequencies": {"cat": 15, "dog": 1}},
+        },
+        {
+            "column_name": "int_column",
+            "column_type": "int",
+            "column_dtype": "int32",
+            "column_stats": {
+                "nan_count": 0,
+                "nan_prop": 0.0,
+                "min": 0,
+                "max": 8,
+                "mean": 4.05,
+                "median": 4.5,
+                "std": 2.60516,
+                "histogram": {"hist": [2, 2, 3, 1, 2, 5, 1, 1, 3, 0], "bin_edges": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]},
+            },
+        },
+        {
+            "column_name": "int_nan_column",
+            "column_type": "int",
+            "column_dtype": "int32",
+            "column_stats": {
+                "nan_count": 6,
+                "nan_prop": 0.3,
+                "min": 0,
+                "max": 8,
+                "mean": 4.71429,
+                "median": 5.0,
+                "std": 2.64367,
+                "histogram": {"hist": [1, 1, 2, 0, 1, 4, 1, 1, 3, 0], "bin_edges": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]},
+            },
+        },
+        {
+            "column_name": "float_column",
+            "column_type": "float",
+            "column_dtype": "float32",
+            "column_stats": {
+                "nan_count": 0,
+                "nan_prop": 0.0,
+                "min": 0.1,
+                "max": 9.9,
+                "mean": 4.585,
+                "median": 4.9,
+                "std": 3.56197,
+                "histogram": {
+                    "hist": [5, 1, 3, 0, 1, 1, 3, 1, 2, 3],
+                    "bin_edges": [0.1, 1.08, 2.06, 3.04, 4.02, 5.0, 5.98, 6.96, 7.94, 8.92],
+                },
+            },
+        },
+        {
+            "column_name": "float_nan_column",
+            "column_type": "float",
+            "column_dtype": "float32",
+            "column_stats": {
+                "nan_count": 8,
+                "nan_prop": 0.4,
+                "min": 0.2,
+                "max": 9.9,
+                "mean": 5.09167,
+                "median": 4.9,
+                "std": 3.87497,
+                "histogram": {
+                    "hist": [3, 0, 2, 0, 1, 1, 0, 0, 2, 3],
+                    "bin_edges": [0.2, 1.17, 2.14, 3.11, 4.08, 5.05, 6.02, 6.99, 7.96, 8.93],
```

Comment:
could we have more tests on the histogram:
- negative values (min=-10, max=-5)
- crossing zero (min=-5, max=5)
- far from zero (min=50,max=60)